rfi_file_monitor/operations/scicataloguer.py

```diff
@@ -479,106 +485,120 @@ def run(self, file: File):
-            sourceFolderHost=host_info["sourceFolderHost"],
-            # TO DO - this would normally be extracted from instrument preferences
-            # instrumentId=str(self.instr_dict["id"]),
-            owner=self.params.owner,
-            contactEmail=self.params.email,
-            orcidOfOwner=self.params.orcid,
-            ownerGroup=self.params.owner_group,
-            accessGroups=access_groups,
-            techniques=[{"name": self.params.technique}],
-            creationTime=(
-                datetime.fromtimestamp(date_method).strftime(
-                    "%Y-%m-%dT%H:%M:%S.%f"
-                )[:-3]
-                + "Z"
-            ),
-        )
-
-        # Add in raw/derived specific variables
-        if self.params.derived_dataset:
+        # Modify base into raw or derived payload
+        if params.derived_dataset:
+            default_payload.type = "derived"
             payload = DerivedPayload(**default_payload.dict())
-            payload.type = "derived"
-            payload.investigator = self.params.investigator
-            payload.inputDatasets = self.params.input_datasets.split(",")
-            payload.usedSoftware = self.params.used_software.split(",")
+            payload = self.is_derived_payload(payload)
         else:
             payload = RawPayload(**default_payload.dict())
-            # TO DO this usually comes from instr dict
-            # Temporary change to fetch from text box for now
-            payload.creationLocation = str(self.params.instrument_choice)
-            payload.principalInvestigator = self.params.investigator
-            payload.endTime = payload.creationTime
-            payload.dataFormat = data_format
-
-        # Add in Directory specific payload details
-        if isinstance(file, Directory):
-            payload.datasetName = (
-                self.params.experiment_name
-                + "/"
-                + str(file.relative_filename.parts[-1])
-            )
-            payload.size = file._total_size
-            payload.numberOfFiles = len(file._filelist)
-
-            # Scientific metadata
-            scientificMetadata: Dict[str, Dict[str, str]] = {}
-            payload.scientificMetadata = (
-                PayloadHelpers.scientific_metadata_concatenation(
-                    scientificMetadata, payload.scientificMetadataDefaults
-                )
-            )
+            payload = self.is_raw_payload(payload, data_format)
 
-        elif isinstance(file, RegularFile):
+        # Remove unneeded metadata defaults
+        del payload.scientificMetadataDefaults
+        return payload
 
-            # Creation of standard file items
-            payload.datasetName = (
-                self.params.experiment_name
-                + "/"
-                + str(PurePosixPath(file.relative_filename))
+    # Adds file specific payload data
+    def is_file_payload(self, _pl, file):
+        _pl.datasetName = (
```

Comment:
pedantic I know, but could we change _pl to _payload, to keep with verbosity of the style.